### PR TITLE
Fixes the 'pan' event for Android devices with the 'touchmove' bug.

### PR DESCRIPTION
**Pull request by jcf120**
Fix adapted from:
http://uihacker.blogspot.co.uk/2011/01/android-touchmove-event-bug.html

### CODE REVIEW

**Comment by jcf120** [comment]
Not all versions of Android have this bug, so it'd probably be neater to test the webview/browser version before applying the fix. I've not looked into how to do this yet, or researched exactly which webview/browser version fixed the bug.

File: src/TouchGestures.js [hunk 1]
```diff
@@ -258,6 +262,9 @@
             if (event === "tap") {
                 this.CONFIG.PREVENT_DBL_CLICK = true;
             }
+            if (event === "pan" && navigator.userAgent.match(/Android/i)) {
```
Comment (shortercode) [inline src/TouchGestures.js]: cache the isAndroid value, we should only have to do this check once

**Comment by shortercode** [comment]
Yeah just been reading into it, total pain in the ass bug by the looks of
it... the hotfix looks mostly okay, I'm a bit worried about stopping the
default behavior further down the chain. Say if you had a pan listener on a
large piece of page, then you wouldn't be able to select any text fields or
links in that chunk of the page.

File: src/TouchGestures.js [hunk 2]
```diff
@@ -112,6 +113,9 @@
                 motion.pos_u_delta = 0;
                 timer = setTimeout(timeout, config.HOLD_MINTIME);
                 that.event('touchstart', motion);
+                if (config.ANDRD_FIX_TOUCH_START_PREVENT_DFLT && event.type === 'touchstart') {
+                    event.preventDefault();
```
Comment (shortercode) [inline src/TouchGestures.js:118]: will prevent links etc from working if they are a child of this element

File: src/TouchGestures.js [hunk 3]
```diff
@@ -19,6 +19,7 @@
                 SWIPE_MINSPEED: 1,
                 END_ON_LEAVE: false,
                 PREVENT_DBL_CLICK: false,
+                ANDRD_FIX_TOUCH_START_PREVENT_DFLT: false,
```
Comment (shortercode) [inline src/TouchGestures.js]: Add comment with link to bug description

**Comment by shortercode** [comment]
Okay from the description of this bug it sounds like android is cancelling touches to scroll. I think if my memory serves correctly you can cancel a scroll using preventDefault on touchmove, but only if its the first touchmove event from that touch.

So that would give us the alternative to prevent scrolling instead of clicks/selects... Not a great trade off.

**Comment by jcf120** [comment]
Good point. I could look into incorporating a solution similar to [android-swipe](https://github.com/TNT-RoX/android-swipe-shim). Seems a bit heavy, but I've not had any better ideas. Thoughts?

**Comment by jcf120** [comment]
Or I suppose we could go with your alternative, and then implement our own scrolling should we ever need it...

**Comment by shortercode** [comment]
Yeah I had a look at that lib as well, interesting solution but it seemed like a bit of a disgusting hack!

**Comment by shortercode** [comment]
Not totally happy with anything we've managed to come up with, probably need to sift through the hammer.js source code and see if they came up with a fix for it. 

Implementing scrolling is a total pain...

**Comment by jcf120** [comment]
I think hammer.js might skirt around the issue by listening for touchmove on the window rather than the element.

**Comment by shortercode** [comment]
mmm if we listened to the global event and filtered the target would that solve our problem? It's something I've considered in the past, as we're a little heavy on the event listeners.

**Comment by jcf120** [comment]
Worth a shot. I'll give it a go.

**Comment by shortercode** [comment]
Cool, as I don't have a device to test on can you check to see if it solves the issue? I'm just blocking out a basic touch source class. Might help with mouse abstraction as well I suppose

**Comment by jcf120** [comment]
Sure, are you gonna push it to master?

**Comment by shortercode** [comment]
I'll branch and add it a separate class, let you review, then we can tie it in later when we're happy with it.

**Comment by jcf120** [comment]
Ah cool, thanks for the repo add. Might as well close this request then.

**Comment by shortercode** [comment]
No worries, I've been meaning to make some changes and it'll be good to have someone else on board.

**Pull request merged**